libnetwork/driver_plugin.py

```diff
@@ -348,10 +352,8 @@ def create_endpoint():
 
     app.logger.info("Creating endpoint %s", endpoint_id)
 
-    # Extract relevant data from the Network data.
-    network_data = get_network_data(network_id)
-    gateway_cidr4, _ = get_gateway_pool_from_network_data(network_data, 4)
-    gateway_cidr6, _ = get_gateway_pool_from_network_data(network_data, 6)
+    # Raise error if they don't have network data stored for this network
+    get_network_data(network_id)
```

Comment:
I'd remove this - it just does an extra etcd query that we don't need to do anymore.